service.onmute/service.py

```diff
@@ -0,0 +1,215 @@
+    def isSubtitle(self):
+        state = xbmc.getCondVisibility('VideoPlayer.SubtitlesEnabled') == 1
+        log("isSubtitle = " + str(state))
+        return state
+         
+         
+    def setSubtitle(self, state):
+        log("setSubtitle = " + str(state))
+        self.Player.showSubtitles(state)
+         
+         
+    def hasSubtitle(self):
+        state = xbmc.getCondVisibility('VideoPlayer.HasSubtitles') == 1
+        log("hasSubtitle = " + str(state))
+        return state
+
+        
+    def isMute(self):
+        state = xbmc.getCondVisibility('Player.Muted') == 1
+        log("isMute = " + str(state))
+        return state
+
+        
+    def setMute(self, state):
+        log("setMute = " + str(state))
+        bailout = 0
+        while self.isMute() != bool(state) and bailout < 13:
+            bailout += 1
+            json_query = '{"jsonrpc":"2.0","method":"Application.SetMute","params":{"mute":%s},"id":1}' %str(state).lower()
+            sendJSON(json_query)
+            xbmc.sleep(500)
+            
+              
+    def getCC(self):
+        # save user parsecaptions settings prior to change
+        state = False
+        json_query = '{"jsonrpc":"2.0","method":"Settings.GetSettingValue","params":{"setting":"subtitles.parsecaptions"}, "id": 1}'
+        json_responce = loadJson(sendJSON(json_query))
+        if 'result' in json_responce and 'value' in json_responce:
+            state = json_responce['result']['value']
+        log("getCC = " + str(state))
+        return state
+        
+    
+    def setCC(self, state):
+        # change parsecaptions setting
+        log("setCC = " + str(state))
+        json_query = '{"jsonrpc":"2.0","method":"Settings.SetSettingValue","params":{"setting":"subtitles.parsecaptions","value":%s}, "id": 1}'%str(state).lower()
+        sendJSON(json_query)
+        
+    
+    def start(self):
+        # enable closed caption if user enables.
+        if self.Monitor.enableCC == True and self.getCC() == False:
+            self.setCC(True)
+            
+        while not self.Monitor.abortRequested():
+            if self.Player.isPlayingVideo() == True:
+                log("autoSub = " + str(self.autoSub))
+                log("subFound = " + str(self.Player.subFound))
+                
+                if xbmcgui.Window(10000).getProperty("PseudoTVRunning") == "True":
+                    self.searchSub = self.Monitor.searchSub
+                    self.Monitor.searchSub = False
+                else:
+                    self.Monitor.searchSub = self.searchSub
+                
+                '''
+                    missing subs return false positive isSubtitle = True,
+                    compare with hasSubtitle to detect true status.
+                '''
+                # check if user has subs already enabled.
+                if (self.isSubtitle() + self.hasSubtitle()) > 1 and self.isMute() == True and self.autoSub == False:
+                    continue
```

Comment:
The reason I commented on that specific line is that the `continue` will start over your loop (not executing the rest below) hence it will **never** reach the `waitForAbort` at the end.

As I said, I don't know how often this case will actually result in `True` and call `continue`.
Assuming it can multiple times (within a few seconds or even worse for a longer period of time) you have a loop bashing away on your cpu (causing high cpu loads and on slow hardware like e.g. a raspberry it *might* even significantly slow the whole system).